stdlib/public/TensorFlow/CompilerRuntime.swift

```diff
@@ -1257,47 +1255,6 @@ fileprivate func setAttrShapeList(
   }
 }
 
-@usableFromInline
-class _ThreadLocalState {
-  var deviceScopes: [(kind: DeviceKind, index: UInt)?] = []
-
-  private static let key: pthread_key_t = {
-    var key = pthread_key_t()
-    pthread_key_create(&key) {
-#if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
-      let _: AnyObject = Unmanaged.fromOpaque($0).takeRetainedValue()
-#else
-      let _: AnyObject = Unmanaged.fromOpaque($0!).takeRetainedValue()
-#endif
-    }
-    return key
-  }()
-
-  var _currentDevice: (DeviceKind, UInt)? {
-    return deviceScopes.last ?? nil
-  }
-
-  @usableFromInline
-  func pushDevice(_ device: (DeviceKind, UInt)?) {
-    deviceScopes.append(device)
-  }
-
-  @usableFromInline
-  func popDevice() {
-    internalConsistencyCheck(deviceScopes.popLast() != nil)
-  }
-
-  @usableFromInline
-  static var value: _ThreadLocalState {
-    if let state = pthread_getspecific(key) {
-      return Unmanaged.fromOpaque(state).takeUnretainedValue()
-    }
-    let state = _ThreadLocalState()
-    pthread_setspecific(key, Unmanaged.passRetained(state).toOpaque())
-    return state
-  }
-}
-
 @usableFromInline
 @_cdecl("_swift_tfc_OpSetDeviceFromScope")
 func _TFCOpSetDeviceFromScope(_ op: CTFEOp, _ status: CTFStatus) {
```

Comment:
Actually, this function is also obsolete. It's only used by GPE earlier. Existing callers can call `TFE_OpSetDevice` directly.

Reply:
It looks like this function is currently used in `tensorflow/swift-bindings`, so I opened a separate ticket to track this work: https://bugs.swift.org/browse/TF-531.